File: antson.java/src/test/java/io/odysz/anson/jprotocol/AnsonResp.java
package io.odysz.anson.jprotocol;

import java.util.ArrayList;
import java.util.HashMap;

import io.odysz.anson.AnsonResultset;

/**Anson message response body
 * @author odys-z@github.com
 */
public class AnsonResp extends AnsonBody {

	private String m;
	private ArrayList<AnsonResultset> rs;
	private HashMap<String, Object> map;

	public AnsonResp() {
		super(null, null);
	}

	public AnsonResp(AnsonMsg<? extends AnsonResp> parent) {
		super(parent, null);
	}

	public AnsonResp(AnsonMsg<? extends AnsonResp> parent, String txt) {
		super(parent, null);
		this.m = txt;
	}

	public AnsonResp(String txt) {
		super(null, null);
		this.m = txt;
	}

	public String msg() { return m; }

	public AnsonBody rs(AnsonResultset rs) {
		if (this.rs == null)
			this.rs = new ArrayList<AnsonResultset>(1);
		this.rs.add(rs);
		return this;
	}

	public ArrayList<AnsonResultset> rs() { return this.rs; }

	public AnsonResultset rs(int ix) {
		return this.rs == null ? null : this.rs.get(ix);
	}

	public AnsonResp data(HashMap<String, Object> props) {
		this.map = props;
		return this;
	}
	
	public HashMap<String, Object> data () {
		return map;
	}
}

File: antson.csharp/antson.csharp/test/io.odysz/jprotocol/AnsonResp.cs
using io.odysz.module.rs;
using System.Collections.Generic;

namespace io.odysz.semantic.jprotocol
{
	/// <summary>Anson message response body</summary>
	/// <author>odys-z@github.com</author>
	public class AnsonResp : AnsonBody
	{
		protected internal string m;

		protected List<AnResultset> rs;

		public Dictionary<string, object> map;

		public AnsonResp() : base(null, null)
		{
		}

		public AnsonResp(AnsonMsg parent) : base(parent, null)
		{
		}

		public AnsonResp(AnsonMsg parent, string txt) : base(parent, null)
		{
			this.m = txt;
		}

		public AnsonResp(string txt) : base(null, null)
		{
			this.m = txt;
		}

		public virtual string Msg()
		{
			return m;
		}

		public virtual AnsonResp Rs(AnResultset rs)
		{
			if (this.rs == null)
			{
				this.rs = new List<AnResultset>(1);
			}
			this.rs.Add(rs);
			return this;
		}

		/// <summary>Add a resultset to list.</summary>
		/// <param name="rs"/>
		/// <param name="totalRows">total row count for a paged query (only a page of rows is actually in rs).
		/// 	</param>
		/// <returns>this</returns>
		public virtual AnsonResp Rs(AnResultset rs, int totalRows)
		{
			if (rs == null)
			{
				this.rs = new List<AnResultset>();
			}

			// rs.total = totalRows;
			this.rs.Add(rs);
			return this;
		}

		public virtual AnsonBody Rs(List<AnResultset> rsLst)
		{
			rs = rsLst;
			return this;
		}

		// public virtual List<AnResultset> Rs () { return rs; }

		public virtual AnResultset Rs(int ix)
		{
			return rs == null ? null : rs[ix];
		}

		public virtual AnsonResp Data(Dictionary<string, object> props)
		{
			map = props;
			return this;
		}

		public virtual Dictionary<string, object> Data()
		{
			return map;
		}
	}
}
